src/QsCompiler/Transformations/Utilities.cs

```diff
@@ -55,6 +55,30 @@ public static class Utilities
                 QsNullable<Tuple<QsPositionInfo, QsPositionInfo>>.Null
             );
 
+        // All of the resolved types of the given expression must match.
+        public static TypedExpression CreateValueArray(params TypedExpression[] expressions)
+        {
+            ResolvedType type = null;
+            if (expressions.Any())
+            {
+                type = expressions.First().ResolvedType;
+                QsCompilerError.Verify(expressions.All(expr => expr.ResolvedType.Equals(type)));
+            }
+            else
+            {
+                type = ResolvedType.New(ResolvedTypeKind.UnitType);
+            }
+
+            return new TypedExpression
+            (
+                ExpressionKind.NewValueArray(expressions.ToImmutableArray()),
+                TypeArgsResolution.Empty,
+                ResolvedType.New(ResolvedTypeKind.NewArrayType(type)),
+                new InferredExpressionInformation(false, false),
```

Comment:
E.g. this here is not accurate in general.

Reply:
Could you elaborate on this, please?

Reply:
HasLocalQuantumDependency is a sticky bit - it is pessimistic in that if it is set to true, the expression may have dependencies, but if it is false, it is guaranteed not to. 
If any of the items in the array literal you are constructing here have quantum dependencies then the entire array expression is considered to (potentially) have a local quantum dependency.

Reply:
This is being track in a separate task.